Revert "[clang-format] Annotate ::operator and Foo::operator correctly"

PiJoules opened the pull request:
Reverts llvm/llvm-project#164048

This led to a regression in clang-format where a space gets added in between the parameter type and `&`. For example, this

```
::test_anonymous::FunctionApplication& ::test_anonymous::FunctionApplication::operator=(const ::test_anonymous::FunctionApplication& other) noexcept {
```

becomes

```
::test_anonymous::FunctionApplication& ::test_anonymous::FunctionApplication::operator=(const ::test_anonymous::FunctionApplication & other) noexcept {
```

PiJoules committed:
Revert "[clang-format] Annotate ::operator and Foo::operator correctly (#164048)"

This reverts commit 3bb9d4a24e40eea1988f6bdc6a79e7a128a2fad9.

frobtech reviewed (approved):
This is a clear regression. I hope we can get a regression test case added to catch it.

frobtech reviewed (approved):
This is a clear regression. I hope we can get a regression test case added to catch it.

HazardyKnusperkeks commented:
It was that urgent, that you couldn't wait for someone to handle it correctly?

PiJoules commented:
I filed https://github.com/llvm/llvm-project/issues/164866 which includes a regression test.

> @Prabhuk I've reverted your revert. Please open a GitHub issue for your use case, which also failed before #111115 was merged.

I think that's the wrong patch? We reverted https://github.com/llvm/llvm-project/pull/164048.

efriedma-quic commented:
The recent activity here is #164048, then #164670, then a948f25?

Reverting a commit from the 18th on the 22nd is within the policy, although I'd usually prefer to give the author of the patch a little more time to respond before merging.

Every commit has a goal to fix something, but the point of the revert policy is to ensure stability for people closely tracking top of tree.  Maintaining that stability is higher priority than a general bugfix, even if it's fixing an important bug.  If you're causing immediate breakage, the right response is to back out, then reland when you have a handle on the regressions.  The fact that the bugfix is fixing a regression from a year ago isn't really relevant; keeping stability is the highest priority.

----

Not sure what we want to do for the branch; we can maybe take a bit more time there to figure out the right approach, once we know what the fix for top-of-tree looks like.

owenca commented:
> As per @efriedma-quic's comment I'll go ahead and revert the commit again to prevent the immediate breakage.

@PiJoules, I don't know if @efriedma-quic's comment was an endorsement of your original hasty (and incomplete) revert, but it didn't address the first part of the policy I quoted above. Please stop forcing your incomplete revert in without first consulting the clang-format team @mydeveloperday, @HazardyKnusperkeks, etc. If we reach the consensus to fix #160513 by reverting instead of by fixing forward, you need to identify the proper "set of patches" to revert and open a pull request for approval by the clang-format team. That set must include both #111115 and #164048 and most likely one or more commits in between. cc @JohnC32

efriedma-quic commented:
> The recent activity here is https://github.com/llvm/llvm-project/pull/164048, then https://github.com/llvm/llvm-project/pull/164670, then https://github.com/llvm/llvm-project/commit/a948f25142e21e60594c6d24e2851f1ff3f216be?

As far as I can tell, what happened is @owenca committed a patch on Oct 17, which broke a workflow for @PiJoules on Oct 21.  He asked for a immediate fix/revert, which did not happen, then he reverted on Oct 22.  After that point, this turned into a revert war.  Currently, the Oct 17 patch is in.

There's also some discussion of a patch which landed last year, but from my perspective it's not relevant.

----

I apologize for not being 100% explicit about what I expected to happen in my previous message.

In the interest of resolving the immediate breakage quickly without further devolving into revert wars, I'm going to be extremely explicit here:

@owenca You have 48 hours from now to resolve the issue @pijoules reported against main.  Either revert to the state before Oct 17, or fix the issue.  If you don't solve it within 48 hours from this message (deadline is 10pm pdt Monday Oct 27), you're not allowed to touch any code related to this issue without getting approval from @PiJoules or me.  Please acknowledge this as soon as you see it.

@PiJoules Don't do anything for 48 hours.  After the deadline, you can revert to the state before Oct 17.

-----

After this is resolved, let's have a Discourse discussion to figure out how we can avoid future revert wars.
